### PR TITLE
Modified sanctions

### PR DESCRIPTION
**Pull request by antonypap**
Seg fault seems to appear in some instances (stages.go:117, `AgentPruneMapping()`)

### CODE REVIEW

**Review by MattSScott — APPROVED** [review]
Pretty sure we've fixed this now!

File: pkg/infra/teams/team3/leader_strat.go [hunk 1]
```diff
@@ -117,3 +117,38 @@ func allocateRandomly(iterator commons.Iterator[state.Item], ids []commons.ID, l
...
+	for id, agent := range agentMap {
+		// Compare to 50 in order to sanction
+		toSanctionOrNot := rand.Intn(100)
+		if toSanctionOrNot > a.Sanctioning(agent) {
+			prunned[id] = agent
+		}
+	}
+	// fmt.Println(len(agentMap))
+	// fmt.Println(len(prunned))
+	return prunned
```
Comment (MattSScott) [inline pkg/infra/teams/team3/leader_strat.go]: Bit annoying, but can we change 'prunned' to 'pruned'

**Comment by MattSScott** [comment]
Great stuff - well done guys

**Review by MattSScott — APPROVED** [review]
Good to merge imo

File: pkg/infra/teams/team3/election_strat.go [hunk 2]
```diff
@@ -85,76 +80,71 @@ func (a *AgentThree) HandleElectionBallot(baseAgent agent.BaseAgent, param *deci
...
+		// check stats of last round from previous level (?)
+		if i == a.fightRoundsHistory.Len()-1 {
+			// search for our agent in fight list and assign action
+			agentsAttackedIDs := res.AttackingAgents()
+			itr2 := agentsAttackedIDs.Iterator()
+			for !itr2.Done() {
 				_, attackingAgentID := itr2.Next()
 				if attackingAgentID == baseAgent.ID() {
 					agentFought = true
+					break
```
Comment (MattSScott) [inline pkg/infra/teams/team3/election_strat.go]: This, and `itr3` perform basically the same functionality, just with different arguments. I would package this into a function

**Review by MattSScott — CHANGES_REQUESTED** [review]
Some cleaning to do. Can we also review my contribution to refactoring? I can't comment on this objectively

**Comment by antonypap** [comment]
Refactor LGTM

**Comment by MattSScott** [comment]
Lovely

**Pull request merged**